src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityWorldAccelerator.java

```diff
@@ -139,72 +105,102 @@ public void addToolUsages(ItemStack stack, @Nullable World world, List<String> t
 
     @Override
     protected long getMaxInputOutputAmperage() {
+        // Take in 8A so that we have a little extra room for cable loss/gaining a buffer in TE mode.
+        // Could change this to 5A for random tick mode, but I think it's not very important.
         return 8L;
     }
 
+    protected long getRandomTickModeAmperage() {
+        return 3L;
+    }
+
+    protected long getTEModeAmperage() {
+        return 6L;
+    }
+
     @Override
     public void update() {
         super.update();
         if (!getWorld().isRemote) {
-            if (isPaused) {
-                if (isActive) {
-                    setActive(false);
-                }
-                return;
-            }
-            if (energyContainer.getEnergyStored() < energyPerTick) {
-                if (isActive) {
-                    setActive(false);
+            if (isPaused && isActive) {
+                setActive(false);
+            } else if (!isPaused) {
+                int currentTick = FMLCommonHandler.instance().getMinecraftServerInstance().getTickCounter();
+                if (currentTick != lastTick) { // Prevent other tick accelerators from accelerating us
+                    lastTick = currentTick;
+                    boolean wasSuccessful = isTEMode() ? handleTEMode() : handleRandomTickMode();
+                    if (!wasSuccessful) {
+                        setActive(false);
+                    } else if (!isActive) {
+                        setActive(true);
+                    }
                 }
-                return;
             }
-            if (!isActive) {
-                setActive(true);
+        }
+    }
+
+    private boolean handleTEMode() {
+        long energyUsage = getTEModeAmperage() * GTValues.V[getTier()];
+        if (energyContainer.getEnergyStored() < energyUsage) return false;
+        energyContainer.removeEnergy(energyUsage);
```

Comment:
This energy consumption logic could be pulled out to `update()`, and performed for both the TE and RandomTick mode. Means slightly less duplicate code here.